### PR TITLE
Intro to ros

### PR DESCRIPTION
**Pull request by RobinAmsters**
Added an introductory text as a preface to the Tutorials. Originally written by Peter Aerts.

### CODE REVIEW

File: tutorials/0_What_is_ROS.ipynb [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<img src=\"images/ros_logo.png\" width=\"200\" />\n",
+    "\n",
+    "# What is ROS ?\n",
+    "\n",
+    "ROS is an abbreviation for '*Robot Operating System*' and is **open source** robotics middleware, which means it is a collection of software frameworks for robotic software development. It provides the services you would expect from an operating system, including hardware abstraction, low-level device control, implementation of commonly-used functionality, message-passing between processes, and package management which can be used to create robot applications.\n",
+    "\n",
+    "The primary goal of ROS is to support the reuse of code. ROS is a distributed framework of processes that enables executables to be individually designed and loosely coupled. These processes are called **Nodes** and can be grouped into **Packages** and **Stacks**. This makes sharing and distributing code easy. ROS also supports a system of code **Repositories** to collaborate on projects and distribute them. Github is commonly used for these collaborations. This allows you to more easily manage and maintain your projects.\n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: ROS Stacks was part of `rosbuild`. I donot think this is required for modern `catkin`-ized packages.

Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: ROS **does not** support repositories. Users _host_ code on repositories like GitHub but they might as well distribute code by an archive file hosted on a webserver running on their personal computer or send the code printed on paper to the end user who types it into the system. GitHub is first and foremost a version control system! So I think saying ROS supports repositories is very misleading.

File: tutorials/0_What_is_ROS.ipynb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    " \n",
+    "One can consider ROS to consist of 5 practical facets:\n",
+    "<img src=\"images/ros_facets.png\" width=\"500\" />\n",
+    "\n",
+    "- ROS enables modular software development by providing a library of free and reusable code packages. This makes robot development faster, easier and more reliable.\n",
+    "- It provides a Run-Time Environment which supports near Real-Time communication between systems elements and data sharing. \n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: Emphasise **near** real time communication. Readers should not confuse ROS with RTOS.

File: tutorials/0_What_is_ROS.ipynb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    "- ROS itself is not a strict standard of programming, but it provides conventions for creating and using ROS supported code in a repeatable and reliable way. For instance, ROS encourages the use of the same data structure to send data between different modules. \n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: I think saying _ROS itself is not a strict standard of programming_ is misleading. ROS is not a standard! But they do provide style guides for programming.

File: tutorials/0_What_is_ROS.ipynb [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    "The master in turn tells the subscriber node that this topic is published at localhost:4321. Nextn the publisher and subscriber establish a connection in order to exchange data. \n",
+    "Take note that the master does not handle any information being sent. It only initializes the connection between publisher and subscriber. \n",
+    "\n",
+    "This publish/subscribe model is many-to-many one-way communication. This is not appropriate for **request/reply** interactions which are often required in a distributed system. Such a **request/reply** communication in ROS is achieved via a **service**. A service is defined by a pair of messages, one for request and one for reply. A node can register a specific service with the ROS master just as it registers/subscribes to its messages. \n",
+    "\n",
+    "For example, let's say we have a <u>Camera node</u> which collects data from a camera and we have an <u>Image processing node</u> to process the images. This situation is depicted in the figure below. \n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: The HTML tag for underline is not parsed correctly.

File: tutorials/0_What_is_ROS.ipynb [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    "ROS is certainly not the only robotics middleware available. Other options are:\n",
+    "- Urbi\n",
+    "- MIRO\n",
+    "- YARP\n",
+    "- OpenRDK\n",
+    "- etc.\n",
+    "\n",
+    "We will not go into detail on the structure and advantages of other available robotics middleware compared to ROS. We do not want to diminish their reputation or influence the reader on their choice of robotics middleware. Instead we will focus on the advantages and possibilities ROS has to offer. \n",
+    "\n",
+    "- Since nodes are connected via a distributed message system, only the required nodes need to be called which can deminish the overall program size. This modularity also has the advantage that, if a component crashes, the whole system doesn’t crash. For example, your robot can still continue its actions even if some sensors have died. \n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: typo: Deminish

File: tutorials/0_What_is_ROS.ipynb [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    "\n",
+    "- ROS is designed for a Linux operating system, however there is the possibility to install and use ROS on macOS and Microsoft Windows. Although these versions are experimental and will run on-top of the OS subsystem for Linux, they are supported by the ROS community. \n",
+    "\n",
+    "- ROS isn’t limited to one programming language. Nodes can be written in some of the most commonly used programming environments. You can write nodes in C++, Python, java and LISP.  Note that C++ offers the most support and that ROS works with Python 2 and is not yet converted to Python 3. \n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: Rephrase: is not fully converted to -> has not been ported to

File: tutorials/0_What_is_ROS.ipynb [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    "\n",
+    "- Through the  ROS community and the fact that ROS is open source, we can avoid the need to develop proprietary code for many aspects of robot control. Troubleshooting via the ROS community can be quick and effective. This makes the development of projects faster and easier. The tutorials on the wiki pages  help new users get comfortable with ROS, butcan sometimes be to technical or don’t explain enough. However, the community of ROS can answer all questions and help you on your way.\n",
```
Comment (ratking82) [inline tutorials/0_What_is_ROS.ipynb]: typo: butcan
typo: to technical -> too technical
Observation: We should avoid embellishing sentences. I think it should be just _can be technical_.

**Review by ratking82 — CHANGES_REQUESTED** [review]
They following are some general recommendation. These are not mandatory but I think will improve the overall discourse. :smile: 

# General recommendations
- Run the text through a spellchecker. I have marked a few. Recommend using the spellchecker nbextension.
- I think **services** should be explained in a section. The provided example does not explain *why* we need to use services.
- The "why" section should come first in my opinion and not at the end.

File: tutorials/0_What_is_ROS.ipynb (hunk 4 above)
Comment (RobinAmsters) [reply tutorials/0_What_is_ROS.ipynb]: It seems to render fine on my machine and the VM ?

Comment (ratking82) [reply tutorials/0_What_is_ROS.ipynb]: I think it is not rendering on the GitHub viewer for some reason.

**Review by ratking82 — APPROVED** [review]
Except for the underline HTML tag not rendering in the built in GitHub viewer, the rest looks okay. :+1:

**Pull request merged**